gravitee-apim-console-webui/src/management/api/general-info/api-general-info-migrate-to-v4-dialog/api-general-info-migrate-to-v4-dialog.component.html

```diff
@@ -0,0 +1,158 @@
+<!--
+
+    Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+            http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<ng-container>
+  <h2 mat-dialog-title class="title">Migrate API to v4</h2>
+
+  @if (migrationResponse$ | async; as response) {
+    <mat-dialog-content>
+      @switch (response.state) {
+        @case ('MIGRATABLE') {
+          @switch (migrationStep) {
+            @case ('INITIAL_CHECK') {
+              <div class="icon-container icon-container--success">
+                <mat-icon svgIcon="gio:check-circled-outline"></mat-icon>
+              </div>
+              <div class="migratable-state">
+                <h3 class="migratable-state__header">Migration ready</h3>
+                <div>Your API is compatible with v4 and can be migrated safely</div>
+              </div>
+            }
+            @case ('CONFIRMATION') {
+              <div class="confirmation-state">
+                <div class="confirmation-state__header">Ready to migrate API to v4</div>
+                <gio-banner-warning class="confirmation-state__warning">
+                  Analytics history will not be preserved
+                  <div gioBannerBody>
+                    Historical analytics data from v2 will not be available in the v4 API. Currently analytics will reset after migration.
+                  </div>
+                </gio-banner-warning>
+                <mat-checkbox (change)="isConfirmed = $event.checked">
+                  <b>I understand that analytics history will be lost</b>
+                </mat-checkbox>
+              </div>
+            }
+          }
+        }
+        @case ('IMPOSSIBLE') {
+          <div class="icon-container icon-container--error">
+            <mat-icon svgIcon="gio:error"></mat-icon>
```

Comment:
What do you think about adding the class `.icon-container--error` to the mat-icon to simplify the CSS and avoid `mat-icon` selector ?